### DISCUSSION TITLE
Frequent OOM of valhalla europe docker service

### DISCUSSION THREAD

**Discussion by akki401** [valhalla/valhalla#4947] [Q&A]
Hello Team,

I have docker valhalla services for continents( Europe, North-America and Asia) which are running on EC2(i4i.2xlarge).
Out of three container services I frequently observed Europe docker container facing OOM and I need to restart the container every time.
The other 2(North-America, Asia) hardly receive traffic but major incoming traffic goes to Europe container.
**I see this issue when any user hit the API trace_attributes.**
I have no idea why that service goes to OOM and how to fix it.
The last input that it served before it goes to OOM tried again after restart and it worked properly.
Find the attached valhalla config and OOM screenshot.
How can I debug when I face OOM. Any clue?
Any suggestion that I need to improve in valhalla config ?
Please help me
![OutOfMemory](https://github.com/user-attachments/assets/43872e71-f3d6-4389-8cc6-cb2d3211f4b2)
[valhalla.json](https://github.com/user-attachments/files/17505167/valhalla.json)

**Comment by chrstnbwnkl** [comment]
The map matcher uses a fine gridded location cache, which can easily explode RAM with large map matching requests. There are config params in the `meili` section of the config that give you some control over the max cache size. 

FWIW, the mid-term goal is to get rid of that cache entirely and use the location search used for route/sources_to_targets instead.
